raiden/api/rest.py

```diff
@@ -1040,7 +1040,7 @@ def initiate_payment(
                 status_code=HTTPStatus.PAYMENT_REQUIRED,
             )
 
-        if payment_status.payment_done is False:
```

Comment:
Hmm nice catch @offerm !

It's really bad this was not caught by our automated test suite. We lack a test that goes into this unhappy case for the rest api it seems.

Would it be too much to ask of you to write one for this PR?

I guess somewhere here: https://github.com/raiden-network/raiden/blob/master/raiden/tests/integration/api/test_restapi.py#L1019

Also this is a bug that will probably hit users so as a user-facing bug, it will need a changelog entry. Can you make one? I will write up a small issue for it.

Reply:
Will Do.

Reply:
Done. 

@LefterisJP I don't think this should go into changelog. The bug introduced only in '0.100.2-rc4'. Let me know if you disagree.

Reply:
@offerm Thank you. Yeah I know it's introduced in rc4, but rc4 is well on its way to become a release, even with this bug in (it's not a critical one). So I would say that if it does become one, we should have a changelog entry.